tests/test_manifest.py

```diff
@@ -86,6 +90,101 @@ def test_manifest_with_anchor(tmpdir, monkeypatch):
     assert manifest.enable_build_targets('bar') == []
 
 
+def test_manifest_with_anchor_and_postfix(tmpdir, monkeypatch):
+    yaml_file = tmpdir / 'test.yml'
+    yaml_file.write_text(
+        """
+.base_depends_components: &base-depends-components
+  depends_components:
+    - esp_hw_support
+    - esp_rom
+    - esp_wifi
+
+examples/wifi/coexist:
+  <<: *base-depends-components
+  depends_components+:
+    - esp_coex
+  depends_components-:
+    - esp_rom
+""",
+        encoding='utf8',
+    )
+
+    manifest = Manifest.from_file(yaml_file)
+    assert manifest.depends_components('examples/wifi/coexist') == ['esp_coex', 'esp_hw_support', 'esp_wifi']
+
+    yaml_file.write_text(
+        """
+.base: &base
+  enable:
+    - if: IDF_VERSION == "5.2.0"
+    - if: IDF_VERSION == "5.3.0"
+
+foo:
+  <<: *base
+  enable+:
+    - if: IDF_VERSION == "5.2.0"
+      temp: true
+    - if: IDF_VERSION == "5.4.0"
+      reason: bar
+""",
+        encoding='utf8',
+    )
+    s_manifest_dict = parse(yaml_file)
+
+    yaml_file.write_text(
+        """
+foo:
+    enable:
+        -   if: IDF_VERSION == "5.2.0"
+        -   if: IDF_VERSION == "5.3.0"
+        -   if: IDF_VERSION == "5.4.0"
+            reason: bar
+""",
+        encoding='utf8',
+    )
+    with open(yaml_file) as f:
+        manifest_dict = yaml.safe_load(f) or {}
+
+    assert s_manifest_dict['foo'] == manifest_dict['foo']
+
+    yaml_file.write_text(
+        """
+.base: &base
+  enable:
+    - if: IDF_VERSION == "5.3.0"
+
+foo:
+  <<: *base
+  enable+:
+    - if: IDF_VERSION == "5.2.0"
+      temp: true
+    - if: IDF_VERSION == "5.4.0"
+      reason: bar
+""",
+        encoding='utf8',
+    )
+
+    s_manifest_dict = parse(yaml_file)
+
+    yaml_file.write_text(
+        """
+foo:
+    enable:
+        -   if: IDF_VERSION == "5.3.0"
+        -   if: IDF_VERSION == "5.2.0"
+            temp: true
+        -   if: IDF_VERSION == "5.4.0"
+            reason: bar
+""",
+        encoding='utf8',
+    )
+    with open(yaml_file) as f:
+        manifest_dict = yaml.safe_load(f) or {}
+
+    assert s_manifest_dict['foo'] == manifest_dict['foo']
+
+
```

Comment:
could you add one more test?

I think the code could have some potential bug. For example

```yaml
.test: &test
  depends_components:
    - a
    - b

foo:
  <<: *test
  depends_components+:
    - c

bar:
  <<: *test
  depends_components+:
    - d
```

will `bar` have both `a, b, c, d`?

This test case is actually testing if the reference of the `test` got modified or not.

since both `foo` and `bar` are sharing the same object of the `depends_components` (you may check the object by `assert id(foo['depends_components']) == id(bar['depends_components'])`), we need to add this test case to make sure the keys won't affect each other.